synapse/http/proxyagent.py

```diff
@@ -223,16 +257,43 @@ def _http_proxy_endpoint(proxy, reactor, **kwargs):
     if proxy is None:
         return None
 
-    # currently we only support hostname:port. Some apps also support
-    # protocol://<host>[:port], which allows a way of requiring a TLS connection to the
-    # proxy.
```

Comment:
This comment still seems valid?

Reply:
Sort of. We support `HTTP_PROXY` and `HTTPS_PROXY`, which does allow us to require a TLS connection to the proxy. But that's done via two separate environment variables, instead of specifying the `protocol` to a single env var.

Because of that I'm not sure if we still needed to include the bit about TLS.

Reply:
I think the pertinent point here is that there are different formats for this env var and we only support a subset of formats.

Reply:
That's fair. I'll try to work that back in!

Reply:
0c5a5b9 is hopefully a bit clearer?

Reply:
Thanks! It just feels a bit sad to lose that information when its already there

Reply:
Yeah I'd missed the underlying point of the comment, assuming that it was old and just referring to the lack of being able to use TLS with a proxy :slightly_smiling_face: